include/boost/multiprecision/cpp_dec_float.hpp

```diff
@@ -56,6 +57,58 @@ namespace backends {
 template <unsigned Digits10, class ExponentType = std::int32_t, class Allocator = void>
 class cpp_dec_float;
 
+namespace detail {
+
+   constexpr std::uint32_t a029747_maker_of_upper_limit(std::uint32_t value)
+   {
+      // Sloanes's A029747: Numbers of the form 2^k times 1, 3 or 5.
+      // m = 140000; Select[Union@Flatten@Outer[Times, {1, 3, 5}, 2^Range[0, Floor[Log2[m]]]], # < m &]
+      // In addition, crop the list to begin with 32 or higher.
+      return ((value <=    32UL) ?    32UL :
```

Comment:
Chris, listen, I'm being seriously nitpicking here, so feel free to ignore this ;)

I was going to suggest this would be better written as a std::lower_bound, but that's not constexpr (at least not in C++11), an alternative would be to manually spell this out as a binary search which would fix the number of comparisons to 5 (I think!) per call.  On the other hand if smaller arguments are more common, then I'm talking nonsense and you should leave it as it is and move on!!